qa/tasks/caasp.py

```diff
@@ -0,0 +1,100 @@
+'''
+Task that deploys a CAASP cluster on all the nodes
+Linter:
+    flake8 --max-line-length=100
+'''
+import logging
+from teuthology.misc import (
+    all_roles_of_type,
+    append_lines_to_file,
+    get_file
+    )
+from teuthology.task import Task
+from util import (
+    get_remote_for_role,
+    )
+log = logging.getLogger(__name__)
+
+
+class Caasp(Task):
+    """
+    Deploy a Caasp cluster on all remotes using Skuba.
+    Nodes are declared in suites/caasp folder
+    """
+
+    def __init__(self, ctx, config):
+        super(Caasp, self).__init__(ctx, config)
+        log.debug("beginning of constructor method")
+        self.ctx['roles'] = self.ctx.config['roles']
+        self.log = log
+        self.remotes = self.cluster.remotes
+        self.mgmt_remote = get_remote_for_role(self.ctx, "skuba_mgmt_host.0")
+        self.ssh_priv = 'caasp_key.rsa'
+        self.ssh_pub = 'caasp_key.rsa.pub'
+        self.set_agent = "eval $(ssh-agent) && ssh-add ~/{} && ".format(
+            self.ssh_priv)
+
+    def __ssh_setup(self):
+        """ Generate keys on management node. Copy pub to all of them. """
+        log.debug("Executing SSH setup")
+        self.__ssh_gen_key()
+        self.__ssh_copy_pub_to_caasp()
+
+    def __ssh_gen_key(self):
+        self.mgmt_remote.run(args=[
+            'ssh-keygen',
+            '-t',
+            'rsa',
+            '-b',
+            '2048',
+            '-N',
+            '',
+            '-f',
+            '{}'.format(self.ssh_priv),
```

Comment:
This code is overkill for simple remote command.
It is hard to read and easy to make a mistake when modifying it.
Can you use here more human readable form, like?
```
self.mgmt_remote.sh('ssh-keygen -t rsa -b 2048 -N "" -f %s' % self.ssh_priv)
```

Reply:
Actually I followed the style used in the salt.py task  which tends to use run() all the time.
I does not look so wrong to me

Reply:
yes, I know salt.py is not ideal, it is hard to maintain this kind way we code, so could you please that people can read and not have to decode it.